src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1894,15 +1895,35 @@ public SyntaxNode GetTypeOfVariableDeclarator(SyntaxNode node)
         public SyntaxNode GetValueOfEqualsValueClause(SyntaxNode node)
             => ((EqualsValueClauseSyntax)node)?.Value;
 
-        public bool IsExecutableBlock(SyntaxNode node)
+        public bool IsPureBlock(SyntaxNode node)
             => node.IsKind(SyntaxKind.Block);
 
+        public bool IsExecutableBlock(SyntaxNode node)
+            => node.IsKind(SyntaxKind.Block, SyntaxKind.SwitchSection);
```

Comment:
interesting.  it looks like this changes an existing method that likely already has usages.  I'm not opposed to that.  but we'll need tests for those features that now involve switch-sections to make sure things are ok.

Reply:
Thanks for catching this. There was actually a crash in "move declaration near reference"! That was because I didn't update FindInnermostCommonExecutableBlock a few lines below. Previously the refactoring didn't show up in switch sections, now it does. There's also a minor formatting issue that was fixed for RemoveUnusedVariable inside switch sections.

➡️ Fixed in 3a5d4b9b89106d55c749a921cc226c8c93cc7f0c